devices/temp_nix.go

```diff
@@ -8,13 +8,46 @@ import (
 
 	"github.com/anatol/smart.go"
 	"github.com/jaypipes/ghw"
+	"github.com/jaypipes/ghw/pkg/block"
 	"github.com/shirou/gopsutil/host"
 )
 
+var disks []block.Disk
+var smDevices []smart.Device
+
 func init() {
 	devs() // Populate the sensorMap
+	RegisterStartup(startBlock)
 	RegisterTemp(getTemps)
 	RegisterDeviceList(Temperatures, devs, defs)
+	RegisterShutdown(endBlock)
+}
+
+func startBlock(vars map[string]string) error {
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return err
+	}
+	for _, disk := range block.Disks {
+		dev, err := smart.Open("/dev/" + disk.Name)
+		if err == nil {
+			disks = append(disks, *disk)
+			smDevices = append(smDevices, dev)
+		}
+	}
+	return err
```

Comment:
`err` variable here is always `nil` (due to loop `err` variable is shadowed). Just return `nil` as an indication that the function was completed successfully.